nixos/modules/programs/wayland/hyprlock.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, pkgs, config, ... }:
+
+let
+  cfg = config.programs.hyprlock;
+in
+{
+  options.programs.hyprlock = {
+    enable = lib.mkEnableOption "hyprlock, Hyprland's GPU-accelerated screen locking utility";
+    package = lib.mkPackageOption pkgs "hyprlock" { };
+    hypridlePackage = lib.mkPackageOption pkgs "hypridle" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [
+      cfg.package
+      cfg.hypridlePackage
+    ];
+
+    # Hyprlock needs Hypridle systemd service to be running to detect idle time
```

Comment:
Not necessarily, it can also be used with swayidle. I think having a separate `hypridle` service would be better for composability.

Reply:
So, what I should do is: move the `hypridle` service to a separate module. And enable `services.hypridle.enable` here? 

And I am not familiar with `swayidle` service, and no module for it is available on Nixpkgs AFAIK. I am not sure how swayidle and hypridle can be used interchangeably depending on which one is available.

Reply:
Done, moved hypridle service to a `services.hypridle` module.

For now, `programs.hyprlock` enables `services.hypridle`.